kythe/java/com/google/devtools/kythe/analyzers/java/JavaIndexer.java

```diff
@@ -89,20 +91,36 @@ public static void main(String[] args) throws AnalysisException, IOException {
     }
 
     CompilationDescription desc = null;
+    String compilationPath = compilation.get(0);
     if (!Strings.isNullOrEmpty(config.getIndexPackRoot())) {
       // java_indexer --index_pack=archive-root unit-key
-      desc = new Archive(config.getIndexPackRoot()).readDescription(compilation.get(0));
+      desc = new Archive(config.getIndexPackRoot()).readDescription(compilationPath);
     } else {
-      // java_indexer kindex-file
-      try {
-        desc = IndexInfoUtils.readIndexInfoFromFile(compilation.get(0));
-      } catch (EOFException e) {
-        if (config.getIgnoreEmptyKIndex()) {
-          return;
+      if (compilationPath.endsWith(IndexInfoUtils.KZIP_FILE_EXT)) {
+        // java_indexer kzip-file
+        try {
+          Collection<CompilationDescription> compilationDescriptions =
+              IndexInfoUtils.readKZip(compilationPath);
+          if (compilationDescriptions.size() != 1) {
```

Comment:
Add support for `--ignore_empty_kindex`?  At least add a TODO?

Reply:
done